File: backend/internal/domain/notification.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type NotificationType string

const (
	NotificationTypeProjectMemberCreated NotificationType = "project.member.created"
	NotificationTypeTaskAssigned         NotificationType = "task.assigned"
	NotificationTypeTaskCommentCreated   NotificationType = "task.comment.created"
)

type Notification struct {
	Id            uuid.UUID        `json:"id"`
	UserId        uuid.UUID        `json:"user_id"`
	ActorId       uuid.UUID        `json:"actor_id"`
	ProjectId     uuid.UUID        `json:"project_id"`
	TaskId        *uuid.UUID       `json:"task_id,omitempty"`
	TaskCommentId *uuid.UUID       `json:"task_comment_id,omitempty"`
	Type          NotificationType `json:"type"`
	ReadAt        *time.Time       `json:"read_at,omitempty"`
	CreatedAt     time.Time        `json:"created_at"`
	UpdatedAt     time.Time        `json:"updated_at"`

	Actor       *User        `json:"actor,omitempty"`
	Project     *Project     `json:"project,omitempty"`
	Task        *Task        `json:"task,omitempty"`
	TaskComment *TaskComment `json:"task_comment,omitempty"`
}

File: frontend/src/types/notification.ts
import type { Project } from './project';
import type { Task, TaskComment } from './task';
import type { User } from './user';

export type NotificationType = 'project.member.created' | 'task.assigned' | 'task.comment.created';

export interface Notification {
  id: string;
  user_id: string;
  actor_id: string;
  project_id: string;
  task_id: string | null;
  task_comment_id: string | null;
  type: NotificationType;
  read_at: string | null;
  created_at: string;
  updated_at: string;
  actor: User | null;
  project: Project | null;
  task: Task | null;
  task_comment: TaskComment | null;
}
